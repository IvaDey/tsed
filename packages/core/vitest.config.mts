// @ts-ignore
import {presets} from "@tsed/vitest/presets";
import {defineConfig} from "vitest/config";

export default defineConfig(
  {
    ...presets,
    test: {
      ...presets.test,
      coverage: {
        ...presets.test.coverage,
        thresholds: {
          statements: 97.33,
          branches: 95.59,
          functions: 96.21,
          lines: 97.33
        }
      }
    }
  }
);