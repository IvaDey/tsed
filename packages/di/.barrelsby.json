{
  "directory": ["./src/common", "./src/node"],
  "exclude": ["**/__mock__", "**/__mocks__", "**/*.spec.ts", "localsContainer.ts"],
  "delete": true
}
