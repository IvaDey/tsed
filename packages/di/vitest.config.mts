// @ts-ignore
import {presets} from "@tsed/vitest/presets";
import {defineConfig} from "vitest/config";

export default defineConfig(
  {
    ...presets,
    test: {
      ...presets.test,
      coverage: {
        ...presets.test.coverage,
        thresholds: {
          statements: 98.69,
          branches: 97.18,
          functions: 97.02,
          lines: 98.69
        }
      }
    }
  }
);
