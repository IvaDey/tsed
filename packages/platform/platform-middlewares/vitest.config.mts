// @ts-ignore
import {presets} from "@tsed/vitest/presets";
import {defineConfig} from "vitest/config";

export default defineConfig(
  {
    ...presets,
    test: {
      ...presets.test,
      coverage: {
        ...presets.test.coverage,
        thresholds: {
          statements: 83.81,
          branches: 90.47,
          functions: 63.63,
          lines: 83.81
        }
      }
    }
  }
);