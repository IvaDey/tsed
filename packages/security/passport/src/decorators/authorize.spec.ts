import {Store} from "@tsed/core";
import {Controller, inject} from "@tsed/di";
import {Middleware, UseBefore} from "@tsed/platform-middlewares";
import {Get, getSpec} from "@tsed/schema";

import {Authorize, PassportMiddleware} from "../index.js";

@Middleware()
class AuthoriseBucket {
  use() {}
}

describe("@Authorize", () => {
  it("should store data", () => {
    class Test {
      @Authorize("local", {
        security: {
          test: ["scope"]
        }
      })
      test() {}
    }

    const store = Store.fromMethod(Test, "test");

    expect(store.get(PassportMiddleware)).toEqual({
      method: "authorize",
      options: {
        security: {
          test: ["scope"]
        }
      },
      originalUrl: true,
      protocol: "local"
    });
  });
  it("should store data (without originalUrl)", () => {
    class Test {
      @Authorize("local", {
        security: {
          test: ["scope"]
        },
        originalUrl: false
      })
      test() {}
    }

    const store = Store.fromMethod(Test, "test");

    expect(store.get(PassportMiddleware)).toEqual({
      method: "authorize",
      options: {
        security: {
          test: ["scope"]
        },
        originalUrl: false
      },
      originalUrl: false,
      protocol: "local"
    });
  });
  it("should support inheritance", async () => {
    abstract class AbstractAdminController {
      @Get("/allEntries")
      getAllEntries(): Promise<unknown> {
        return Promise.resolve({});
      }
    }

    @Authorize("loginAuthProvider")
    @Controller("/admin")
    class AdminController extends AbstractAdminController {}

    @Controller("/admin/bucket")
    @UseBefore(AuthoriseBucket)
    class BucketAdminController extends AbstractAdminController {}

    expect(getSpec(AdminController)).toMatchSnapshot();
    expect(getSpec(BucketAdminController)).toMatchSnapshot();
    expect(Reflect.getOwnPropertyDescriptor(AdminController.prototype, "getAllEntries")).toBeDefined();
    expect(Reflect.getOwnPropertyDescriptor(BucketAdminController.prototype, "getAllEntries")).not.toBeDefined();

    const adminController = inject(AdminController);

    expect(await adminController.getAllEntries()).toEqual({});

    expect(Store.fromMethod(BucketAdminController, "getAllEntries").get(PassportMiddleware)).toEqual(undefined);
    expect(Store.fromMethod(AbstractAdminController, "getAllEntries").get(PassportMiddleware)).toEqual(undefined);
    expect(Store.fromMethod(AdminController, "getAllEntries").get(PassportMiddleware)).toEqual({
      method: "authorize",
      options: {},
      originalUrl: true,
      protocol: "loginAuthProvider"
    });
  });
});
