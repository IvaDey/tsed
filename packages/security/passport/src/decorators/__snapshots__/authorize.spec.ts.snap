// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@Authorize > should support inheritance 1`] = `
{
  "paths": {
    "/admin/allEntries": {
      "get": {
        "operationId": "abstractAdminControllerGetAllEntries",
        "parameters": [],
        "responses": {
          "200": {
            "description": "Success",
          },
        },
        "tags": [
          "AdminController",
        ],
      },
    },
  },
  "tags": [
    {
      "name": "AdminController",
    },
  ],
}
`;

exports[`@Authorize > should support inheritance 2`] = `
{
  "paths": {
    "/admin/bucket/allEntries": {
      "get": {
        "operationId": "abstractAdminControllerGetAllEntries",
        "parameters": [],
        "responses": {
          "200": {
            "description": "Success",
          },
        },
        "tags": [
          "BucketAdminController",
        ],
      },
    },
  },
  "tags": [
    {
      "name": "BucketAdminController",
    },
  ],
}
`;
